### PR TITLE
#484: fix reservation views and tests

### PR DESCRIPTION
**Pull request by vladarefiev**
PR for https://github.com/timelesslounge/timelessis/issues/484:
- fixed ReservationViews
- fixed all tests
- removed useless tests

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#504](https://github.com/timelesslounge/timelessis/issues/504) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#504](https://github.com/timelesslounge/timelessis/issues/504) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/260136f4-feba-4a23-8449-5b336ef24768); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by vladarefiev** [comment]
@amihaiemil ping %)

File: tests/factories.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ class Meta:
         sqlalchemy_session_persistence = "commit"
 
 
-class ReservationFactory(factory.alchemy.SQLAlchemyModelFactory):
+class ReservationFactory(DictMixin, factory.alchemy.SQLAlchemyModelFactory):
```
Comment (amihaiemil) [inline tests/factories.py]: @vladarefiev what is ``DictMixin`` for?

File: tests/integration/it_reservations_test.py [hunk 2]
```diff
@@ -7,35 +7,29 @@
...
 def test_edit(client):
     reservation_old = factories.ReservationFactory()
-    reservation_new = factories.ReservationFactory()
+    reservation_new = factories.ReservationFactory.get_dict()
     url = url_for("reservations.edit", id=reservation_old.id)
     response = client.post(url, data=reservation_new)
-    assert response.status_code == HTTPStatus.OK
-    assert reservation_new.comment in response.data
+    assert response.status_code == HTTPStatus.FOUND
+    reservation = Reservation.query.get(reservation_old.id)
```
Comment (amihaiemil) [inline tests/integration/it_reservations_test.py]: @vladarefiev one-time variable. I know you like them, but can we rename it to ``edited``? :D

File: timeless/reservations/views.py [hunk 3]
```diff
@@ -18,13 +18,38 @@
...
+    Deletes location using id and redirects to list page
+    """
+    success_view_name = "reservations.list"
+    model = Reservation
+
+
+ReservationsListView.register(BP, "/", name="list")
+ReservationsViewCreate.register(BP, "/create", name="create")
+ReservationsEditView.register(BP, "/edit/<int:id>", name="edit")
+ReservationsDeleteView.register(BP, "/delete/<int:id>", name="delete")
```
Comment (amihaiemil) [inline timeless/reservations/views.py]: @vladarefiev I don't think other endpoints were given a ``name`` when being registered -- what is it for?

File: timeless/reservations/views.py [hunk 4]
```diff
@@ -18,13 +18,38 @@
 
 
 class ReservationsListView(views.ListView):
```
Comment (amihaiemil) [inline timeless/reservations/views.py]: @vladarefiev shouldn't you add a puzzle to secure these views? At the moment I'm not sure they are secured -- as I remember, they should extend ``SecuredView`` :D

File: tests/integration/it_reservations_test.py [hunk 5]
```diff
@@ -7,35 +7,29 @@
...
-        column.name: str(getattr(reservation, column.name))
-        for column in reservation.__table__.columns
-    }
-    url = url_for("reservations.create_reservation")
-    response = client.post(url, data=reservation_dict)
-    assert response.status_code == HTTPStatus.CREATED
+    reservation = factories.ReservationFactory.get_dict()
+    url = url_for("reservations.create")
+    response = client.post(url, data=reservation)
+    assert response.status_code == HTTPStatus.FOUND
```
Comment (amihaiemil) [inline tests/integration/it_reservations_test.py]: @vladarefiev in the case of creating an entity, the server should return ``201 CREATED``, not ``FOUND``. Can you change that or is it too much work?

File: tests/integration/it_reservations_test.py [hunk 6]
```diff
@@ -7,35 +7,29 @@
 from timeless.restaurants.models import Reservation
 
 
-@pytest.mark.skip("reservations.list does not implemented yet")
 def test_list(client):
     reservation = factories.ReservationFactory()
     response = client.get(url_for("reservations.list"))
     assert response.status_code == HTTPStatus.OK
-    assert reservation.comment in response.data
+    assert reservation.comment in response.data.decode('utf-8')
```
Comment (amihaiemil) [inline tests/integration/it_reservations_test.py]: @vladarefiev can you also assert that the response contains only 1 reservation and not more?

Comment (vladarefiev) [reply tests/integration/it_reservations_test.py]: @amihaiemil ok

File: tests/factories.py (hunk 1 above)
Comment (vladarefiev) [reply tests/factories.py]: @amihaiemil to get a object converted to dict

File: tests/integration/it_reservations_test.py (hunk 5 above)
Comment (vladarefiev) [reply tests/integration/it_reservations_test.py]: @amihaiemil no, you are wrong. Here is a templates, no `201` is here. particularly this view are redirecting you if success

File: tests/integration/it_reservations_test.py (hunk 2 above)
Comment (vladarefiev) [reply tests/integration/it_reservations_test.py]: @amihaiemil it's two time :D . ok, no problem ;)

File: timeless/reservations/views.py (hunk 4 above)
Comment (vladarefiev) [reply timeless/reservations/views.py]: @amihaiemil I don't know, I don't implement these views in this PR, just fix them. Let's ask @emilianodellacasa

File: timeless/reservations/views.py (hunk 3 above)
Comment (vladarefiev) [reply timeless/reservations/views.py]: @amihaiemil It's for providing a names in namespace for them. Sometimes when people didn't it, therefore I'm fixing their tests right now ;)

**Comment by vladarefiev** [comment]
@amihaiemil fixed, check it out pls ;)

File: tests/integration/it_reservations_test.py (hunk 5 above)
Comment (amihaiemil) [reply tests/integration/it_reservations_test.py]: @vladarefiev I see, ok

**Comment by amihaiemil** [comment]
@vladarefiev ok, good to merge from my point of view. @emilianodellacasa please see our question about securing views above :D

File: timeless/reservations/views.py (hunk 4 above)
Comment (emilianodellacasa) [reply timeless/reservations/views.py]: @vladarefiev @amihaiemil there is already a puzzle to secure views, let's not create duplicates

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 25 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#504](https://github.com/timelesslounge/timelessis/issues/504) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
The job [#504](https://github.com/timelesslounge/timelessis/issues/504) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**